server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/controllers/DatasetController.java

```diff
@@ -546,7 +561,15 @@ public Dataset uploadFiles(@Parameter(name = "Dataset Id from data service", req
 
         // check if the dataset is hazard dataset
         String dataType = dataset.getDataType();
-        isHazardDataset = HazardConstants.DATA_TYPE_HAZARD.contains(dataType);
+        String subDataType;
+        if (dataType.contains(":")) {
+            // Compare what comes after the name space (e.g. probabilisticEarthquakeRaster from ergo:probabilisticEarthquakeRaster)
+            subDataType = dataType.split(":")[1];
+        } else {
+            subDataType = dataType;
+        }
+        isHazardDataset = HazardConstants.DATA_TYPE_HAZARD.stream().anyMatch(s1 -> s1.contains(subDataType));
```

Comment:
Just asking, this doesn't get bothered by case sensitivity, right?

Reply:
@longshuicy @ylyangtw What do you think about this? I can make this case insensitive, but should I? Right now this fixes it so whether it's ergo:deterministicEarthquake or ncsa:deterministicEarthquake it will properly increment/decrement the users number of datasets while we finalize on the type. I can also make it match for the case someone uploads ergo:DeterministicEarthquake, but is that going to far? In theory, at some point, the semantic service should prevent wrong data types, but we are not enforcing it.

Reply:
i think we should leave it as it is. The only change I would envision for the future is moving that semantic definition from ncsa space to "ergo/incore" space once they are ready. Not the name of that definition though.

Reply:
@ywkim312 could you take another look and if good could you approve and merge?

Reply:
For the current purpose, based on your comment above, I think that it should be okay. I will approve and merge.